### PR TITLE
Add calculator

### PR DESCRIPTION
**Pull request by see-why**
This pull request introduces a basic calculator application, including functionality for evaluating mathematical expressions, rendering results in a formatted box, and unit tests to ensure correctness. The most important changes are grouped below by theme.

### Core Functionality:

* Added a `Calculator` class in `pkg/calculator.py` to evaluate infix mathematical expressions with support for `+`, `-`, `*`, and `/` operators. It includes methods for token parsing, operator precedence handling, and evaluation logic.

### Application Entry Point:

* Created a `main.py` file to serve as the entry point for the calculator application. It parses command-line arguments, evaluates expressions using the `Calculator` class, and prints the results.

### Result Rendering:

* Added a `render` function in `pkg/render.py` to format the expression and result into a visually appealing box for display.

### Testing:

* Implemented unit tests in `tests.py` to validate the functionality of the `Calculator` class, covering basic operations, complex expressions, edge cases, and error handling.

### CODE REVIEW

File: calculator/calculator/main.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        print('Example: python main.py "3 + 5"')
+        return
+
+    expression = " ".join(sys.argv[1:])
+    try:
+        result = calculator.evaluate(expression)
+        to_print = render(expression, result)
+        print(to_print)
+    except Exception as e:
+        print(f"Error: {e}")
```
[the remark this answers is not in the document]
Comment (see-why) [reply calculator/calculator/main.py:22]: I'll skip this at this time.
